components/transactions/PaymentBreakdown.tsx

```diff
@@ -1,70 +1,62 @@
 import * as React from 'react';
 import { SearchBar, ListItem, Button, Icon, Overlay } from 'react-native-elements';
-import {View} from './Themed';
+import {View} from '../Themed';
 import {StyleSheet, Text, FlatList} from 'react-native';
 import { TextInput} from 'react-native-gesture-handler';
 import {connect} from 'react-redux';
-import {Contact} from '../redux/types/types.Contact';
-import { ContactTransactionPair, PaymentStatus } from '../redux/types/types.ContactTransactionPair';
-import {AppState} from '../redux/root-reducer';
+import {Contact} from '../../redux/types/types.Contact';
+import { TransactionStatus, PaymentStatus } from '../../redux/types/types.TransactionStatus';
+import {AppState} from '../../redux/root-reducer';
 import {Dispatch} from 'redux';
-import { Transaction } from '../redux/types/types.Transaction';
-import { addMultipleContactTransactionPairs, editAmount, removeContactFromTransaction, removeContactsByTransactionId } from '../redux/contactTransactionPair/contactTransactionPair.action';
-import * as lodash from 'lodash';
-
+import { Transaction } from '../../redux/types/types.Transaction';
+import { addMultipleTransactionStatus, editTransactionStatusAmount, removeTransactionStatus, removeTransactionStatusByTransactionId } from '../../redux/transactionStatus/transactionStatus.action';
+import _ from 'lodash';
 
 interface StateProps {
-    contactTransactionPairs: ContactTransactionPair[],
+    contactTransactionPairs: TransactionStatus[],
     contacts: Contact[]
 }
 
 interface DispatchProps {
-    addMultipleContactTransactionPairs: Function,
-    removeContactFromTransaction: Function,
-    removeContactsByTransactionId: Function,
+    addMultipleTransactionStatuses: Function,
+    removeStatusFromTransaction: Function,
+    removeStatusesByTransactionId: Function,
     editAmount: Function 
 }
 
 interface ParentComponentProps {
     currentTransaction: Transaction,
     editable: boolean,
-    saveChanges: boolean,
+    updateContacts: boolean,
 }
 
 interface PaymentBreakdownProps extends StateProps, DispatchProps, ParentComponentProps {}
 
 interface PaymentBreakdownState{
     search: string,
     searchResultList: Contact[],
-    contactTransactionPairs: (ContactTransactionPair)[],
+    transactionStatuses: (TransactionStatus)[],
     contactViewVisible: boolean,
     editable: boolean,
-    saveChanges: boolean,
+    updateContacts: boolean,
 }
 
-// const actions = {
-//     addContactToTransaction: (contact: Contact, transaction: Transaction): any => true,
-//     removeContactFromTransaction: (contactId: string): any => true,
-//     removeContactsByTransactionId: (transactionIdl: string) : any => true,
-//     editAmount: (contactId: string, amount: number): any => true
-// }
-
 class PaymentBreakdown extends React.Component<PaymentBreakdownProps, PaymentBreakdownState> {
```

Comment:
Yeah this is on my list of to-dos. This component existed since when I still wrote everything as class components, so the logic is a bit lifecycle-oriented. I refactored it minimally just so that I can get a working Create Transaction screen first.

Reply:
We should try our best to keep things consistent since most of the components are functional. Try to change it to functional and if it doesn't work, we can merge it and then fix it later.